src/core/fetcher.ts

```diff
@@ -2,24 +2,32 @@ import Item from "./Item";
 import Dependency from "./Dependency";
 import { StatusBar } from "../ui/status-bar";
 import {
-  versions as ciVersions
+  sparseIndexServerURL,
+  versions as sparseVersions
+} from "../api/sparse-index-server";
+import {
+  versions as cratesVersions
 } from "../api/crates-index-server";
 import compareVersions from "../semver/compareVersions";
 import { CompletionItem, CompletionItemKind, CompletionList, workspace, window } from "vscode";
 import { sortText } from "../providers/autoCompletion";
+import { CrateMetadatas } from "../api/crateMetadatas";
 
 export function fetchCrateVersions(dependencies: Item[]): [Promise<Dependency[]>, Map<string, Dependency[]>] {
-  StatusBar.setText("Loading", "👀 Fetching crates.io");
-
   // load config
   const config = workspace.getConfiguration("");
   const shouldListPreRels = !!config.get("crates.listPreReleases");
-  const indexServerURL = config.get<string>("crates.indexServerURL");
-  let versions = ciVersions;
-  let transformer = transformServerResponse(versions, shouldListPreRels);
-  if (!indexServerURL) {
-    window.setStatusBarMessage("Crates index server URL is not configured. Looking local index", 2000);
+  const indexServerURL = config.get<string>("crates.indexServerURL") ?? sparseIndexServerURL;
+
+  StatusBar.setText("Loading", "👀 Fetching " + indexServerURL.replace(/^https?:\/\//, ''));
+
+  let versions;
+  if (indexServerURL.startsWith(sparseIndexServerURL)) {
```

Comment:
> I am thinking that if we can understand that url is sparse compatible or cis compatible it will be better.

Indeed, it would be better. A valid sparse index should have a https://index.crates.io/config.json file, so the check could be: if `https://<indexServerURL>/config.json` exists (http status 200), it's a sparse compatible index

Reply:
Do you want to implement  or should I plan myself ?

Reply:
I'm actually implementing it :yum: (but if you want to do it no problem)

Reply:
Nope I am pretty happy that someone put effort in it. Thanks and please go on. 🚀